lib/Sema/CSDiagnostics.h

```diff
@@ -1977,6 +1977,16 @@ class MissingQuialifierInMemberRefFailure final : public FailureDiagnostic {
   bool diagnoseAsError();
 };
 
+/// Emits a warning about an attempt to use the 'as' operator as the 'as!'
+/// operator.
+class CoercionAsForceCastFailure final : public ContextualFailure {
```

Comment:
Do we have to do something special so that the type checker doesn't bail out after applying the fix?

Reply:
There's an `IsWarning` on `ConstraintFix` that should be set to `true` for this. It defaults to `false` in the `ConstraintFix` constructor

Reply:
Oops I just realized that this is the `FailureDiagnostic` subclass. `IsWarning` is being set here:

```cpp
  AllowCoercionToForceCast(ConstraintSystem &cs, Type fromType, Type toType,
                            ConstraintLocator *locator)
       : ContextualMismatch(cs, FixKind::AllowCoercionToForceCast, fromType,
                            toType, locator, /*warning*/ true) {}
```

Reply:
Yup, and `applySolution` checks for solutions with all-warning fixes and continues to apply them as normal